### PR TITLE
Updated boids project

### LINKED ISSUE #143: boids not quite working
Hi,

Something does not look right with hosted boids demo.

https://boids.pyviz.demo.anaconda.com/notebooks/boids.ipynb

There are loads of warnings like

`WARNING:param.ParameterizedMetaclass: Use method 'params' via param namespace`

The result is that the final animation is not displaying.

### PR DESCRIPTION
**Pull request by jlstevens**
This PR updates the pinning fixing the param warnings and changes the `VectorField` constructor accordingly. Fixes #143.

This would be ready to merge if I could get `anaconda-project lock` to work: currently it either complains about `nomkl` not being found (which I've tried removing) but even after that I get solve conflicts.

### CODE REVIEW

**Comment by jlstevens** [comment]
@jbednar The boids project didn't originally have an `anaconda-project-lock.yml` but now I've added one.

The changes I had to make to the `anaconda-project.yml` are [here](https://github.com/pyviz-topics/examples/pull/145/commits/ee467f560193cde4ac9781d48b69ee4ba6fd7ea6#diff-86f486ab397bce37ff84c48f2f7cedff1e46a6b9072912fcc26a7310d58c3baa).

However, I'm not sure it is worth having to add `conda-forge` as a channel in order to be able to generate the lockfile. In addition, the tests [are no longer passing right now](https://github.com/pyviz-topics/examples/pull/145/checks?check_run_id=2493730847#step:6:869) as apparently `pytest` can no longer be found?

**Comment by jlstevens** [comment]
I've solved the `anaconda-project lock` issue: there seems to be a regression since 0.8.2 such that 0.9.0 and 0.9.1 no longer seems to support multiple entries in the `env_spec`.

File: boids/anaconda-project.yml [hunk 1]
```diff
@@ -41,5 +40,10 @@ env_specs:
...
     packages:
-      - nbsmoke=0.2.8
-      - pytest=4.4.1
+    - nbsmoke=0.2.8
+    - pytest=4.4.1
+
+platforms:
+  - linux-64
+  - win-64
+  - osx-64
```
Comment (jbednar) [inline boids/anaconda-project.yml:49]:
```suggestion
  - osx-64

```

**Review by jbednar — APPROVED** [review]
Looks good, thanks!

**Pull request merged**